### PR TITLE
Make an mdspan unit test compile on GPU

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

File: tests/test_mdspan_ctors.cpp [hunk 1]
```diff
@@ -45,20 +45,37 @@
...
 namespace stdex = std::experimental;
 _MDSPAN_INLINE_VARIABLE constexpr auto dyn = stdex::dynamic_extent;
 
+
+void test_mdspan_ctor_data_carray() {
+  size_t* result = allocate_array<size_t>(6);
+  dispatch([=] _MDSPAN_HOST_DEVICE () {
+    int data[1] = {42};
+    stdex::mdspan<int, stdex::extents<1>> m(data);
+    result[0] = m.data() == data ? 1 : 0;
```
Comment (dalg24) [inline tests/test_mdspan_ctors.cpp]:
```C++
#define POOR_MAN_ASSERT_EQ(LHS, RHS) \
if (!(LHS == RHS)) {
  printf("expected equality of %s and %s", #LHS, #RHS); // can improve to print values when possible
  ++errors; // increment some error counter
}

Comment (mhoemmen) [reply tests/test_mdspan_ctors.cpp]: @dalg24 I have a vague memory that the Sierra (Sandia's finite-element suite) developers had written something like this.

Comment (crtrott) [reply tests/test_mdspan_ctors.cpp]: addressed

File: tests/offload_utils.hpp [hunk 2]
```diff
@@ -1,6 +1,13 @@
 
 namespace {
 bool dispatch_host = true;
+
+#define __MDSPAN_DEVICE_ASSERT_EQ(LHS, RHS) \
+if (!(LHS == RHS)) { \
+  printf("expected equality of %s and %s", #LHS, #RHS); \
```
Comment (dalg24) [inline tests/offload_utils.hpp]:
```suggestion
  printf("expected equality of %s and %s\n", #LHS, #RHS); \
```

File: tests/offload_utils.hpp [hunk 3]
```diff
@@ -1,6 +1,13 @@
 
 namespace {
 bool dispatch_host = true;
+
+#define __MDSPAN_DEVICE_ASSERT_EQ(LHS, RHS) \
```
Comment (dalg24) [inline tests/offload_utils.hpp:5]: Double underscore prefix doesn't sound like a good idea.  I don't think it is legal.

**Review by dalg24 — DISMISSED** [review]
Make sure you add the `'\n'` and you can merge as far as I am concerned

File: tests/offload_utils.hpp (hunk 3 above)
Comment (crtrott) [reply tests/offload_utils.hpp:5]: OK but then we need to rename a few others too. We can do that in a separate PR.

**Review by dalg24 — APPROVED** [review]

**Pull request merged**